superset/datasets/dao.py

```diff
@@ -147,14 +147,22 @@ def validate_metrics_uniqueness(dataset_id: int, metrics_names: List[str]) -> bo
 
     @classmethod
     def update(
-        cls, model: SqlaTable, properties: Dict[str, Any], commit: bool = True
+        cls,
+        model: SqlaTable,
+        properties: Dict[str, Any],
+        commit: bool = True,
     ) -> Optional[SqlaTable]:
         """
         Updates a Dataset model on the metadata DB
         """
 
         if "columns" in properties:
-            cls.update_columns(model, properties.pop("columns"), commit=commit)
+            cls.update_columns(
+                model,
+                properties.pop("columns"),
+                commit=commit,
+                override_columns=bool(properties.get("override_columns")),
+            )
 
         if "metrics" in properties:
             cls.update_metrics(model, properties.pop("metrics"), commit=commit)
```

Comment:
for put /dataset, does override_columns literally mean just the columns? Are we putting or patching here? Does it include the dataset? Wouldn't we be also overriding the metrics?

Reply:
from sqllab, we don't allow users to create metrics so if you look at the `update_metrics` function we end up just deleting all the rows since `properties["metrics"]` is always None from that context.